documentation/src/main/asciidoc/user_guide/streams.adoc

```diff
@@ -534,5 +539,41 @@ public class WordFrequencyExample {
 
 This way you can still utilize all of the cores locally when calculating the most frequent element.
 
+*Remove specific entries*
+
+Distributed streams can also be used as a way to modify data where it lives.
+For example you may want to remove all entries in your cache that contain
+a specific word.
+
+[source,java]
+----
+public class RemoveBadWords {
+   public static void main(String[] args) {
+      // Lines removed
+      String word = ..
+
+      c1.entrySet().parallelStream()
+         .filter(e -> e.getValue().contains(word))
+         .forEach((c, e) -> c.remove(e.getKey());
+----
+
+If we carefully note what is serialized and what is not, we notice that only the word along
+with the operations are serialized across to other nods as it is captured by the lambda.
+However the real saving piece is that the cache operation is performed on the primary
+owner thus reducing the amount of network traffic required to remove these values from the
+cache. The cache is not captured by the lambda as we provide a special BiConsumer method
+override that when invoked on each node passes the cache to the BiConsumer 
+
+One thing to keep in mind using the forEach command in this manner is that the underlying
+stream obtains no locks. The cahe remove operation will still obtain locks naturally, but
+the value could have changed from what the stream saw. That means that the entry could
+have been changed after the stream read it but the remove actually removed it.
+
+We have specifically added a new variant which is called LockedStream which will be
+covered in the next section.
```

Comment:
I suggest "a new variant called LockedStream, which does guarantee that the entries seen by the `forEach` action are up to date, and which is covered in the next section."